### PR TITLE
BaSP-TG-137: Search shared Component

### PR DESCRIPTION
**Pull request by renzopimentel**
Create search bar for tables, making it functional for every table data.

How to test:
Login as an employee and use the search bar in the home section.
In the search bar, you can filter name, last name, project and client name and dates.

### CODE REVIEW

File: src/Components/Employees/Home/index.js [hunk 1]
```diff
@@ -70,6 +70,22 @@ const EmployeesHome = () => {
...
+    : filteredList.filter(
+        (dato) =>
+          dato.name?.toLowerCase().includes(search.toLocaleLowerCase()) ||
+          dato.lastName?.toLowerCase().includes(search.toLocaleLowerCase()) ||
+          dato.email?.toLowerCase().includes(search.toLocaleLowerCase()) ||
+          dato.phone?.toLowerCase().includes(search.toLocaleLowerCase()) ||
+          dato.clientName?.toLowerCase().includes(search.toLocaleLowerCase()) ||
+          dato.startDate?.toLowerCase().includes(search.toLocaleLowerCase()) ||
+          dato.endDate?.toLowerCase().includes(search.toLocaleLowerCase()) ||
+          dato.employees?.toLowerCase().includes(search.toLocaleLowerCase())
```
Comment (CristianBucci) [inline src/Components/Employees/Home/index.js:86]: You can add `dato. description?`

File: src/Components/Shared/Table/Table.js [hunk 2]
```diff
@@ -1,15 +1,37 @@
...
+    : data.filter(
+        (dato) =>
+          dato.name?.toLowerCase().includes(search.toLocaleLowerCase()) ||
+          dato.lastName?.toLowerCase().includes(search.toLocaleLowerCase()) ||
+          dato.email?.toLowerCase().includes(search.toLocaleLowerCase()) ||
+          dato.phone?.toLowerCase().includes(search.toLocaleLowerCase()) ||
+          dato.clientName?.toLowerCase().includes(search.toLocaleLowerCase()) ||
+          dato.startDate?.toLowerCase().includes(search.toLocaleLowerCase()) ||
+          dato.endDate?.toLowerCase().includes(search.toLocaleLowerCase()) ||
+          dato.employees?.toLowerCase().includes(search.toLocaleLowerCase())
```
Comment (CristianBucci) [inline src/Components/Shared/Table/Table.js:19]: You can add `dato. description?`

File: src/Components/Shared/Table/Table.js [hunk 3]
```diff
@@ -1,15 +1,37 @@
...
 import styles from './Table.module.css';
 import { Link } from 'react-router-dom';
 
 const Table = ({ data, headers, dataValues, setShowModal, location }) => {
+  const [search, setSearch] = useState('');
+
+  const results = !search
+    ? data
+    : data.filter(
+        (dato) =>
```
Comment (CristianBucci) [inline src/Components/Shared/Table/Table.js:11]: You can name it `value` instead of `data`

File: src/Components/Shared/Table/Table.js [hunk 4]
```diff
@@ -1,15 +1,37 @@
...
+
   return (
     <>
       <div className={styles.container}>
         <div className={styles.top}>
           <div className={styles.searchBox}>
             <img src="/assets/images/lens.svg" alt="update" />
-            <input type="text" placeholder="Search"></input>
+            <input
+              type="text"
```
Comment (CristianBucci) [inline src/Components/Shared/Table/Table.js:29]: The type should be `search`

File: src/Components/Shared/Table/Table.js (hunk 2 above)
Comment (renzopimentel) [reply src/Components/Shared/Table/Table.js:19]: done, thanks

**Review by matescarabino — APPROVED** [review]
LGTM

**Review by NicoSan02 — APPROVED** [review]
🕵️‍♂️ QA FEEDBACK

BaSP-TG-137: Search shared Component #58

We wanted to tell you that we consider that your pr complies the requirements to be approved, and that it can be merged into the master branch 👍

**Pull request merged**